src/cpp/src/speculative_decoding/speculative_decoding_eagle_utils.hpp

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <filesystem>
+#include <vector>
+
+#include "openvino/genai/generation_config.hpp"
+
+#include "openvino/runtime/core.hpp"
+#include "openvino/op/constant.hpp"
+
+namespace ov {
+namespace genai {
+namespace speculative_decoding {
+
+constexpr std::size_t DEFAULT_NUM_ASSISTANT_TOKENS = 4;
+
+// Set num_assistant_tokens to default if not specified and validate config
+void ensure_num_assistant_tokens_is_set(ov::genai::GenerationConfig& config);
+
+// Eagle3 runtime configuration
+struct Eagle3RTInfo {
+    bool eagle3_mode = false;
+    std::vector<int> hidden_layers_list;
+    std::filesystem::path dt_mapping_table;
+};
+
+Eagle3RTInfo extract_eagle_mode_from_config(ov::AnyMap& config, const std::filesystem::path& models_path = {});
```

Comment:
Please provide the function description. Also, it doesn't only extract `Eagle3` mode, it also creates vector of layers to use features from. Might be better to rename to `create_eagle_info_from_config()` or something like this.